### ISSUE TITLE
Cannot load configuration file: invalid type: integer `200`, expected a string

### ISSUE THREAD

**Issue by nvuillam** [lycheeverse/lychee#1338]
[Example in documentation](https://lychee.cli.rs/#/usage/config?id=default-location-of-the-configuration-file) contains the following example:

![image](https://github.com/lycheeverse/lychee/assets/17500430/e2b6573f-cddc-431d-98ae-7eb0c9b7856c)

But when using the latest version, we have the following error:

```sh
 Error:  Error while loading config: Cannot load configuration file `/github/workspace/.github/linters/lychee.toml`: Failed to parse configuration file
  
  Caused by:
      TOML parse error at line 32, column 11
         |
      32 | accept = [200, 206, 429]
         |           ^^^
      invalid type: integer `200`, expected a string
```

Out lychee.toml file contains:
```toml
# Comma-separated list of accepted status codes for valid links.
accept = [200, 206, 429]
```

**Comment by mre** [comment]
Oh, okay. We changed the parsing and status codes need to be strings now. The reason is that ranges are also supported now, e.g. `"200..=204"`. Range syntax is identical to the normal Rust range syntax. We should update the docs to reflect that. Want to send a pull request? 
Maybe we could support both, strings and integers, but I don't know how easy that will be.

**Comment by nvuillam** [comment]
@mre I think the best would be to support both in a 0.14.1 version, because if after a version upgrade, existing configuration file triggers a crash, it means it's a breaking change, and according to [semver](https://semver.org/), in that case 0.14.0 should have been 1.0.0   :)

**Comment by mre** [comment]
Supporting both is probably ideal yes.

Just would like to make one small remark that semver allows breaking changes before 1.0. In fact, that's the very reason why lychee is not 1.0 yet: I do anticipate quite a few more breaking changes before the first stable release; it's just that this specific one is probably avoidable. Whether we will support both, string and int, depends on the complexity of the implementation and what side effects this will have down the road. So far I don't see any reason why it shouldn't be possible.

**Comment by nvuillam** [comment]
Indeed you're right about semver, I forgot the "before 1.0.0 rule" :)

But as lychee is more and more used, projects are more and more dependent on it, maybe it would be time to release a 1.0.0 ? 👼

**Comment by Techassi** [comment]
We should be able to enable support for single integers and list of integers by widening our custom serde implementation. I could tackle this is a PR in the next few days.

**Comment by Techassi** [comment]
This will be fixed as soon as #1345 is merged.

**Comment by mre** [comment]
Amazing work!
